src/renderer/display_list.rs

```diff
@@ -152,6 +136,27 @@ impl<'a> DisplayList<'a> {
         }
         Ok(())
     }
+
+    fn get_lineno_width(&self) -> usize {
```

Comment:
imo I would instead make this `get_max_lineno()` which keep this focused and ensure policy around `anonymized_line_numbers` is obvious

Reply:
Just want to confirm you wish it to become

1.

```rust
fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    let max_lineno = self.get_max_lineno();
    let lineno_width = match max_lineno {
        None => 0,
        Some(_max) if self.anonymized_line_numbers => ANONYMIZED_LINE_NUM.len(),
        Some(0) => 1,
        Some(max) => (max as f64).log10().floor() as usize + 1,
   };

    // ...
}
```

2.

```rust
fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    let lineno_width = self.get_max_lineno();
    
    // ...
}
```
3.

```rust
fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    let max_lineno = self.get_max_lineno();
    let lineno_width = get_lineno_width(max_lineno);
    // ...
}
```